### PR TITLE
I8 filtering courses to run

### LINKED ISSUE #8: Run the script based on month entered in the google sheet
The script will run 3 times a year, mostly one month after Fall/Winter/SpringSummer in Feb/June/Oct. 

Fall Term courses could be upto 15
Winter Term courses could be up to 10
Spring/Summer less than 5 

Based on the entry in the Google Spreedsheet columns (`When does script run? (Feb, June, Oct), Script Run?`) and the month configured in environmental variables `SCRIPT_RUN_MONTH` the script will pick courses to run.

```
if (When does script run? (Feb, June, Oct) == SCRIPT_RUN_MONTH & Script Run? == ''):
     pick the courses
else: don't pick up the courses
```

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #8 

Testing File: https://docs.google.com/spreadsheets/d/18mtjJ8RRP5MnSQYqzhJtwVBmdkFPy3m0u_i_Rkmuk0A/edit#gid=0
`.env` file Update: Add `SCRIPT_RUN_MONTH=June`

Change the values in `When does script run? (Feb, June, Oct`) and `Script Run?` and update the `SCRIPT_RUN_MONTH` value in `.env` file appropriately and see script pick up the courses as desired .

If you don't want to run the whole competency update process once courses are picked (for testing only) comments the lines L113 - L116 to see result faster. 

```
rubric_data: Dict[str, Any] = self.getting_rubrics()
        logger.debug(f'Rubric data: {rubric_data}')
        self.filter_df_course_ids.apply(
            lambda course: self.start_competencies_assigning_process(course, rubric_data), axis=1)
```

Test are written to support the use case

### CODE REVIEW

File: ipe_process_orchestrator/orchestrator.py [hunk 1]
```diff
@@ -18,25 +18,28 @@ def __init__(self, props, original_df, api_handler) -> None:
...
 
-    def _clean_up_ipe_dataframe(self) -> None:
+    def filter_course_list_to_run_and_cleanup(self) -> None:
         """
-        Clean up the dataframe
-        1. leading and trailing spaces df.columns
-        2. only courses id list with values that are numbers. Removes Shell, shell, empty, n/a, shell(23333)
-        3. the original dataframe will remain same and the filtered dataframe will be created with courseIds
-
+        1. this filter the courses that need to be run based on the 'Script Run?' and `When does the script run columns
```
Comment (zqian) [inline ipe_process_orchestrator/orchestrator.py]: change 

based on the 'Script Run?' and `When does the script run columns

into 

based on the 'Script Run?' and 'When does script run? (Feb, June, Oct)' columns

File: ipe_process_orchestrator/orchestrator.py [hunk 2]
```diff
@@ -104,7 +107,9 @@ def start_composing_process(self):
         """
         This is the place where all the IPE process flow will be orchestrated.
         """
-        self._clean_up_ipe_dataframe()
+        self.filter_course_list_to_run_and_cleanup()
+        if self.filter_df_course_ids.empty:
```
Comment (zqian) [inline ipe_process_orchestrator/orchestrator.py:111]: add a log message here when the filtered list is empty?

File: ipe_utils/df_utils.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import pandas as pd
+from datetime import datetime
+from constants import (WHEN_TO_RUN_SCRIPT, SCRIPT_RUN, COL_COURSE_ID)
```
Comment (zqian) [inline ipe_utils/df_utils.py:3]: is the () needed in the import statement?

**Review by zqian — COMMENTED** [review]
I have set "SCRIPT_RUN_MONTH=June" configuration variable, and the test of job run was successful.

File: ipe_process_orchestrator/orchestrator.py (hunk 2 above)
Comment (pushyamig) [reply ipe_process_orchestrator/orchestrator.py:111]: I did not add the log statement here but during the filtering step. Below sample of it. I just generalized statement when you have and don't have courses to run. Look [here](https://github.com/tl-its-umich-edu/IPE-competencies-updater/pull/19/files#diff-76281cfa507df5613d7e2e1df0658848ad5188ee6495760bdb32aef8087f595fR40)
```
ipe_process_orchestrator.orchestrator - INFO - 0 courses to run the IPE process: []
```

File: ipe_utils/df_utils.py (hunk 3 above)
Comment (pushyamig) [reply ipe_utils/df_utils.py:3]: That is preferred way for wrapping a long line of imports in Python. This is part of PEP recommendation. https://stackoverflow.com/a/29534096

File: ipe_process_orchestrator/orchestrator.py (hunk 1 above)
Comment (pushyamig) [reply ipe_process_orchestrator/orchestrator.py]: Done

**Comment by pushyamig** [comment]
@zqian I think I resolved all the suggestion. Please review again

**Review by zqian — APPROVED** [review]

**Pull request merged**